be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -93,7 +93,15 @@ Status HorizontalGeneralTabletWriter::reset_segment_writer(bool eos) {
     auto name = gen_segment_filename(_txn_id);
     SegmentWriterOptions opts;
     opts.is_compaction = _is_compaction;
+
+    if (auto metadata = _tablet_mgr->get_latest_cached_tablet_metadata(_tablet_id);
```

Comment:
will it be necessary to update the SegmentWriter in `column_mode_partial_update_handler.cpp` as well?